recipes/armadillo/all/conanfile.py

```diff
@@ -195,89 +190,75 @@ def requirements(self):
-        )
-        self._cmake.definitions["ALLOW_OPENBLAS_MACOS"] = (
-            self.options.use_blas == "openblas"
-        ) and self.settings.os == "Macos"
-        self._cmake.definitions["ALLOW_BLAS_LAPACK_MACOS"] = (
-            self.options.use_blas != "framework_accelerate"
-        )
-        self._cmake.definitions["BUILD_SHARED_LIBS"] = self.options.shared
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
+        tc.cache_variables["DETECT_HDF5"] = self.options.use_hdf5
+        tc.cache_variables["USE_OPENBLAS"] = (self.options.use_blas == "openblas")
+        tc.cache_variables["USE_MKL"] = self.options.use_blas == "intel_mkl" and self.options.use_lapack == "intel_mkl"
+        tc.cache_variables["USE_SYSTEM_LAPACK"] = self.options.use_lapack == "system_lapack"
+        tc.cache_variables["USE_SYSTEM_BLAS"] = (self.options.use_blas == "system_blas")
+        tc.cache_variables["USE_SYSTEM_ATLAS"] = self.options.use_lapack == "system_atlas"
+        tc.cache_variables["USE_SYSTEM_HDF5"] = False
+        tc.cache_variables["USE_SYSTEM_ARPACK"] = self.options.use_arpack
+        tc.cache_variables["USE_SYSTEM_SUPERLU"] = self.options.use_superlu
+        tc.cache_variables["USE_SYSTEM_OPENBLAS"] = False
+        tc.cache_variables["USE_SYSTEM_FLEXIBLAS"] = self.options.use_blas == "system_flexiblas"
+        tc.cache_variables["ALLOW_FLEXIBLAS_LINUX"] = self.options.use_blas == "system_flexiblas" and self.settings.os == "Linux"
+        tc.cache_variables["ALLOW_OPENBLAS_MACOS"] = self.options.use_blas == "openblas" and self.settings.os == "Macos"
+        tc.cache_variables["OPENBLAS_PROVIDES_LAPACK"] = self.options.use_lapack == "openblas"
+        tc.cache_variables["ALLOW_BLAS_LAPACK_MACOS"] = self.options.use_blas != "framework_accelerate"
+        tc.cache_variables["BUILD_SHARED_LIBS"] = self.options.shared
+        tc.cache_variables["BUILD_SMOKE_TEST"] = False
+        tc.generate()
+
+        deps = CMakeDeps(self)
+        deps.generate()
 
     def source(self):
-        tools.get(
+        get(self,
             **self.conan_data["sources"][self.version],
             strip_root=True,
-            destination=self._source_subfolder,
-            filename="{name}-{version}.tar.xz".format(
-                name=self.name, version=self.version
-            ),
+            filename="f{self.name}-{self.version}.tar.xz"
         )
 
-    def _patch_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-
     def build(self):
-        self._patch_sources()
-        cmake = self._configure_cmake()
+        apply_conandata_patches(self)
+        replace_in_file(
+            self,
+            os.path.join(self.source_folder, "CMakeLists.txt"),
+            "include(CheckLibraryExists)",
+            "include(CheckLibraryExists)\nset(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS on)",
```

Comment:
This can be set from the toolchain

Reply:
Has guidance on this changed? I have [previously been advised](https://github.com/conan-io/conan-center-index/pull/6512#discussion_r690343079) to move these statements into the CMakeLists.txt

Reply:
Thats a very old PR :) when we needed the CMake wrapper that was true... Patches are way worse then one line in the recipe